src/net.h

```diff
@@ -96,9 +96,18 @@ class CConnman
         CONNECTIONS_ALL = (CONNECTIONS_IN | CONNECTIONS_OUT),
     };
 
+    struct Options
+    {
+        uint64_t nLocalServices = 0;
+        int nMaxConnections = 0;
+        int nMaxOutbound = 0;
+        int nBestHeight = 0;
+        CClientUIInterface* interface = nullptr;
```

Comment:
ChainParams seems like it could qualify as an option. In the long term I think it would make more sense for the UI to connect to the various backend signals rather than the net layer having any knowledge of the UI, but that change seems outside the scope of this PR.